### ISSUE TITLE
DeprecationWarnings: numpy - elementwise comparison, calling nonzero on 0d arrays, `np.object`.

### ISSUE THREAD

**Issue by IAlibay** [MDAnalysis/mdanalysis#3535]
Couple of numpy deprecation warnings we'll need to take care of.

This gets thrown in `hbond_autocorrel.py`: - done

```
hbond_autocorrel.py:435: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
    pair = np.delete(pair, np.where(pair==exclude), 0)
```

This gets throws in `test_hydrogenbond_autocorrel_deprecated.py`:

```
<__array_function__ internals>:180: DeprecationWarning: Calling nonzero on 0d arrays is deprecated, as it behaves surprisingly. Use `atleast_1d(cond).nonzero()` if the old behavior was intended. If the context of this warning is of the form `arr[nonzero(cond)]`, just use `arr[cond]`.
```

This deprecation warning gets thrown in `test_topologyattr.py`: - done

```
DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    dtype=np.object)
```

MDA version: 2.1.0-dev0

**Comment by jbarnoud** [comment]
One more:

```
MDAnalysisTests/core/test_topologyattrs.py::TestAtomnames::test_next_emptyresidue
MDAnalysisTests/core/test_topologyattrs.py::TestAtomnames::test_prev_emptyresidue
  /home/jon/miniconda3/envs/work/lib/python3.8/site-packages/numpy/testing/_private/utils.py:397: DeprecationWarning: The truth value of an empty array is ambiguous. Returning False, but in future this will result in an error. Use `array.size > 0` to check that an array is not empty.
    if isdesnan and isactnan:
```

**Comment by IAlibay** [comment]
Adding this to 2.2.0 milestone.

**Comment by ojeda-e** [comment]
This looks like a `good first issue`. Would you please confirm @IAlibay? (Pinning you since you opened the issue)

**Comment by IAlibay** [comment]
> This looks like a `good first issue`. Would you please confirm @IAlibay? (Pinning you since you opened the issue)   

Sure sounds good to me

**Comment by IAlibay** [comment]
self assigning this as I'd like to get it done ASAP for 2.2.0
